common/remote/rmtsmtp.cpp

```diff
@@ -773,77 +800,138 @@ static const char *quit="QUIT\r\n";
 
 static void doSendEmail(CMailInfo & info, CMailPart const & part)
 {
-    info.open();
-    StringBuffer outbuff;
-
-    info.read();
-    info.getHelo(outbuff);
-    info.write(outbuff.str(), outbuff.length());
-    info.read();
-
-    info.getMailFrom(outbuff.clear());
-    info.write(outbuff.str(), outbuff.length());
-    info.read();
-
-    unsigned numRcpt = info.numRecipients();
-    for(unsigned i=0; i<numRcpt; ++i)
-    {
-        info.getRecipient(i, outbuff.clear());
-        info.write(outbuff.str(), outbuff.length());
-        info.read();
-    }
-
-    info.write(data, strlen(data));
-    info.read();
-    info.getHeader(outbuff.clear());
-    part.getHeader(outbuff);
-    outbuff.append("\r\n");
-    info.write(outbuff.str(), outbuff.length(), "mail header");
-    part.write(info);
-    info.write(endMail, strlen(endMail), "end of mail body");
-    info.read();
-
-    info.write(quit, strlen(quit));
-    info.read();
+    try
+    {
+        info.open();
+        StringBuffer outbuff;
+
+        info.read(0);
+        info.getHelo(outbuff);
+
+        bool ok;
+        int retries = 2;
+        while (retries >= 0)
```

Comment:
This could be commoned up unto a function for the common case

`info.writeAndRead(buffer, len, numretries)
`

Reply:
ok, yes, cleaner code.